_posts/2019-07-29-plonk-stack.md

```diff
@@ -0,0 +1,211 @@
+
+The motto of OpenFaaS is "Serverless Functions Made Simple" and this is reflected in the project values:
+
+* developers-first
+
+* operationally simple
+
+* community-centric
+
+To date there are dozens of end-user companies, some of which have given permission to list their logo on [OpenFaaS.com](https://openfaas.com), over 230 contributors, 18k GitHub stars, and [hundreds of community blogs and events recorded](https://github.com/openfaas/faas/blob/master/community.md).
+
+If you're new to OpenFaaS and are wondering what it's all about, then you can get up to speed with my video from Goto: Serverless Beyond the Hype:
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/yOpYYYRuDQ0" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+In simple terms OpenFaaS offers:
+
+* an easy way to package any code or binary
+* a rich ecosystem of language templates
+* a function store to collaborate and share
+* metrics, auto-scaling, and dozens of detailed tutorials
+* a native experience on Kubernetes, without the overheads
+* a dedicated community ready to help you, when you need it most
+
+[OpenFaaS.com](https://www.openfaas.com/) is hosted by [OpenFaaS Ltd](mailto:sales@openfaas.com) and developed by a [voluntary team of developers and experts](https://www.openfaas.com/team/).
+
+Users can get commercial support, help with architecture, training and consultation from [OpenFaaS Ltd](mailto:sales@openfaas.com).
+
+#### NATS
+
+> NATS.io is a simple, secure and high performance open source messaging system for cloud native applications, IoT messaging, and microservices architectures.
+
+[NATS](https://nats.io) was developed by [Derek Collison](https://www.linkedin.com/in/derekcollison/) and gained significant traction among Cloud Native developers. So much so, that it was accepted into the CNCF as a hosted project in 2018.
+
+OpenFaaS uses NATS Streaming as a queuing system, so that invocations can be queued up and processed in parallel as capacity becomes available within your cluster. Asynchronous invocations are built-in and do not require any updates to your endpoint, you can even request a HTTP callback when the invocation has completed.
+
+```sh
+faas-cli store deploy figlet
+
+# Synchronous, or blocking
+
+curl http://gateway.example.com/function/figlet -d NATS
+
+# Asynchronous, or non-blocking:
+
+curl http://gateway.example.com/function/figlet -d NATS -H "X-Callback-Url: http://gateway.example.com/function/after-figlet"
+```
+
+NATS, just like Linkerd and OpenFaaS aims to be simple to install, operate, and use over time.
+
+> Derek has now gone on to found a new company called [Synadia](https://synadia.com) aiming to "Connect Everything" through NATS 2.0. NATS 2.0 brings many similar features to those offered by a Service Mesh and is being positioned as a "digital dial-tone".
+
+#### Kubernetes
```

Comment:
This section talks more about OpenFaaS than it does K8s

Reply:
That is true, I'll fix that by giving some context.